src/utils/server-utils.ts

```diff
@@ -141,24 +145,66 @@ async function getWCSCoverage(serverUri: string) {
       'domainSet.temporalDomain.gml:timePosition',
     );
   } catch (error) {
-    throw new Error(
-      `Failed loading dates with request GET: ${requestUri} for WCS Coverage, ${error}`,
-    );
+    console.error(error);
+    return {};
   }
 }
 
 /**
- * Given a WMS or WCS serverUri, return a Map of available dates
+ * Gets the available dates for a groundstation layer
+ *
+ * in layers.json each uri is given a constant, large date range (2000-01-01 -> 2023-12-21) to try get the api to give all data
+ * TODO Once the api is fixed this needs to be fixed as its currently a hacky solution to get around the api's caveats
+ *
+ */
+async function getGroundstationCoverage({
+  data: uri,
+  fallbackData: fallbackUri,
+  beginDate,
+  endDate,
+}: GroundstationLayerProps) {
+  let data;
+  const maxDateQuery = `?beginDateTime=${beginDate}&endDateTime=${endDate}`;
+  try {
+    // eslint-disable-next-line fp/no-mutation
+    data = (await (
+      await fetch(uri + maxDateQuery, { mode: 'cors' })
+    ).json()) as GroundstationLayerData;
+  } catch (ignored) {
+    // eslint-disable-next-line fp/no-mutation
+    data = (await (
+      await fetch(fallbackUri || '')
+    ).json()) as GroundstationLayerData;
+  }
+  return data
+    .map(item => moment.utc(item.date).valueOf())
+    .filter((date, index, arr) => {
+      return arr.indexOf(date) === index;
+    }); // filter() here removes duplicate dates because indexOf will always return the first occurrence of an item
+}
+
+/**
+ * Given a WMS or WCS serverUri, return a Map of available dates.
```

Comment:
Let rephrase this to:

Load available dates for WMS and WCS using serverUri and for GeoJSONs using their API endpoint.